### PR TITLE
Feat: Additional test coverage and component structure for TemplateCard

### PR DESCRIPTION
**Pull request by kalleericson**
## Hey, I just made a Pull Request!

`TemplateCard` in `plugin-scaffolder-react` is too intricate and monolithic. This PR breaks it up into smaller, exposed, components to improve testability and reusability.

Also adds test-selectors to all its important nodes, to allow downstream projects to test the component properly.

**No logic or layout change, strictly code maintenance**

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by kalleericson** [comment]
Hey @benjdlambert ! 

Since the `TemplateCard` takes very limited props, the only real way to differentiate different types or categories of cards in lists of components seem to be to carbon copy the entire Card. Breaking it up like this, would allow downstream projects to reuse a lot of the logic and structure, whilst still making local differentiations. 

This might be swinging a bit too far in the "exposed"-direction, admittedly, but to expose types such as the AdditionalLinks and such would also go a long way in building for consistent downstream cases.

I find that there's a lot of need throughout Backstage for better test selectors. We use a lot of end-to-end tests, and it's currently a pain to make the scripts reliable and non-hacky. So I would argue a selector structure like suggested here would be nice to consider more broadly.  

I figured since it's clearly marked `alpha`, that it lowered the risk of exposing components - even those under development still? If that's a misunderstanding, I'm all ears.

**Comment by acierto** [comment]
Hey @kalleericson,

What stops you to define your own component and supply it via `EXPERIMENTAL_TemplateListPageComponent`?

**Comment by kalleericson** [comment]
Hi @acierto !

Nothing is _stopping_ us from using `EXPERIMENTAL_TemplateListPageComponent`, apart from a general approach to try and use the upstream components to the maximum extent. To carbon the `TemplateCard` _just_ to add test selectors, would add a lot of unnecessary complexity and required sync in upcoming upstream updates.

If this PR is taking the concept of breaking the card down into smaller components too far, I'll be happy to roll back all but the added selectors? A bit reluctantly, considering code complexity, but it would help us out a lot.

**Comment by benjdlambert** [comment]
@kalleericson I'm a little reluctant to break apart this component any further and re-export them as building blocks for other usages. I don't think we're there yet. I'd much rather wait for something like #27726 to settle before exposing smaller pieces of components and patterns.

Happy to move forward with the selectors and tests though :pray:

File: plugins/scaffolder-react/src/next/components/TemplateCard/TemplateCard.tsx [hunk 1]
```diff
@@ -15,81 +15,42 @@
...
+  url: string;
+}
+
 /**
  * The Props for the {@link TemplateCard} component
  * @alpha
  */
 export interface TemplateCardProps {
   template: TemplateEntityV1beta3;
-  additionalLinks?: {
```
Comment (benjdlambert) [inline plugins/scaffolder-react/src/next/components/TemplateCard/TemplateCard.tsx:87]: Can we keep this type inline for now? Don't really wanna add too much more API surface exposed out of the package :pray:

**Review by benjdlambert — CHANGES_REQUESTED** [review]
Thanks for this! Just a small change about the types and I think it's good :Tada:

File: plugins/scaffolder-react/src/next/components/TemplateCard/TemplateCard.tsx (hunk 1 above)
Comment (kalleericson) [reply plugins/scaffolder-react/src/next/components/TemplateCard/TemplateCard.tsx:87]: Sure thing, you're the boss :)

**Review by benjdlambert — APPROVED** [review]
Nice! Thanks for this :tada:

**Pull request merged**